examples/model_selection/plot_grid_search_stats.py

```diff
@@ -0,0 +1,527 @@
+"""
+==================================================
+Statistical comparison of models using grid search
+==================================================
+
+This example illustrates how to statistically compare the performance of models
+trained and evaluated using :class:`~sklearn.model_selection.GridSearchCV`.
+
+"""
+
+print(__doc__)
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+from itertools import combinations
+from math import factorial
+from scipy.stats import t
+from sklearn.datasets import make_moons
+from sklearn.model_selection import GridSearchCV, RepeatedStratifiedKFold
+from sklearn.svm import SVC
+
+# set seaborn parameters
+sns.set_style("darkgrid")
+
+###############################################################################
+# We will start by simulating moon shaped data (where the ideal separation
+# between classes is non-linear), adding to it a moderate degree of noise.
+# Datapoints will belong to one of two possible classes represented
+# by two features. We will simulate 50 samples for each class:
+
+X, y = make_moons(noise=0.352, random_state=1)
```

Comment:
Yes, 100 samples might be too little. The only thing is that I tried re-executing the code with 1000 samples and the statistical results change (e.g. the corrected t-test is now significant) so the interpretation in the text should also be changed (I don't mind changing it though, but just to let you know you will need to proof-read that part again). 

I'm actually wondering now that since p-values depend so much on the sample size, maybe we should also be giving a measure of the effect size in the frequentist approach.